community/detectors/roxy_wi_cve_2022_31137/src/main/java/com/google/tsunami/plugins/detectors/cves/cve202231137/Cve202231137Detector.java

```diff
@@ -0,0 +1,174 @@
+import com.google.tsunami.proto.VulnerabilityId;
+import java.io.IOException;
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+import java.time.Clock;
+import java.time.Instant;
+import java.util.Optional;
+import javax.inject.Inject;
+
+/** A {@link VulnDetector} that detects Roxy-wi RCE CVE-2022-31137. */
+@PluginInfo(
+    type = PluginType.VULN_DETECTION,
+    name = "Roxy-wi RCE CVE-2022-31137 Detector",
+    version = "0.1",
+    description = "This detector checks Roxy-wi RCE (CVE-2022-31137)",
+    author = "am0o0",
+    bootstrapModule = Cve202231137DetectorBootstrapModule.class)
+public final class Cve202231137Detector implements VulnDetector {
+  private static final GoogleLogger logger = GoogleLogger.forEnclosingClass();
+  @VisibleForTesting static final String VULNERABLE_REQUEST_PATH = "app/options.py";
+  private static final String HTTP_PARAMETERS = "alert_consumer=1&ipbackend=\";%s+#";
+
+  private final Clock utcClock;
+  private final HttpClient httpClient;
+
+  private final PayloadGenerator payloadGenerator;
+
+  @Inject
+  Cve202231137Detector(
+      @UtcClock Clock utcClock, HttpClient httpClient, PayloadGenerator payloadGenerator) {
+
+    this.utcClock = checkNotNull(utcClock);
+    this.httpClient = checkNotNull(httpClient);
+    this.payloadGenerator = checkNotNull(payloadGenerator);
+  }
+
+  @Override
+  public DetectionReportList detect(
+      TargetInfo targetInfo, ImmutableList<NetworkService> matchedServices) {
+    logger.atInfo().log("Cve202231137Detector starts detecting.");
+
+    return DetectionReportList.newBuilder()
+        .addAllDetectionReports(
+            matchedServices.stream()
+                .filter(NetworkServiceUtils::isWebService)
+                .filter(this::isRoxyWiWebService)
+                .filter(this::isServiceVulnerable)
+                .map(networkService -> buildDetectionReport(targetInfo, networkService))
+                .collect(toImmutableList()))
+        .build();
+  }
+
+  private boolean isRoxyWiWebService(NetworkService networkService) {
+    // note that this fingerprint phase is only for vulnerable versions,
+    // because newer Roxy-Wi versions have different endpoints for login
+    String targetUrl =
+        NetworkServiceUtils.buildWebApplicationRootUrl(networkService) + "app/login.py";
+    try {
+      Optional<String> response =
+          httpClient.send(get(targetUrl).withEmptyHeaders().build(), networkService).bodyString();
+      return response.isPresent() && response.get().contains("<title>Login page - Roxy-WI</title>");
+    } catch (IOException e) {
+      logger.atWarning().withCause(e).log("Request to target %s failed", networkService);
+    }
+    return false;
+  }
+
+  private boolean isServiceVulnerable(NetworkService networkService) {
+    return isVulnerableWithoutCallback(networkService);
+  }
+
+  private boolean isVulnerableWithoutCallback(NetworkService networkService) {

```

Comment:
Please remove the `isVulnerableWithoutCallback` method and move the logic inside `isServiceVulnerable`

```suggestion
```